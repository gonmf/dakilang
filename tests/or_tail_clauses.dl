actor("chaplin").
athlete('ronaldo').
athlete('messi').
politician("ghandi").
politician("arnold").
politician(123).
actor("arnold").
athlete("arnold").

famous(X) :- actor(X); \
             athlete(X); \
             politician(X).

listing

famous(X)?


other(X) :- abc(X, "gual").
other(X) :- abc(X, "guali").
other(X) :- abc(X, "gual"). # This instruction will produce a warning

other(X) :- abc(X, "guali"); abc(X, "gual").

###########################

parent('john', 'damian').
parent('john', 'ellie').
parent('sarah', 'john').

grandparent(gp, gc) :- parent(gp, parent), parent(parent, gc).

sibling(X, Y) :- grandparent(Z, X), grandparent(Z, Y).

son(X, Y) :- parent(Y, X).

related(person, other) :- son(parent, other); parent(parent, other); sibling(parent, other).

related('sarah', anyone)?

