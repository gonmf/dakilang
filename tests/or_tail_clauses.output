> actor("chaplin").
> athlete('ronaldo').
> athlete('messi').
> politician("ghandi").
> politician("arnold").
> politician(123).
> actor("arnold").
> athlete("arnold").
>
> famous(X) :- actor(X);  athlete(X); politician(X).
>
> listing
 0: actor('chaplin').
 1: athlete('ronaldo').
 2: athlete('messi').
 3: politician('ghandi').
 4: politician('arnold').
 5: politician(123).
 6: actor('arnold').
 7: athlete('arnold').
 8: famous(X) :- actor(X).
 9: famous(X) :- athlete(X).
10: famous(X) :- politician(X).

>
> famous(X)?
X = 'chaplin'

X = 'arnold'

X = 'ronaldo'

X = 'messi'

X = 'ghandi'

X = 123

>
>
> other(X) :- abc(X, "gual").
> other(X) :- abc(X, "guali").
> other(X) :- abc(X, "gual"). # This instruction will produce a warning
Clause already exists

>
> other(X) :- abc(X, "guali"); abc(X, "gual").
>
> %%%%%%%%%%%%%
Syntax error at %%%%%%%%%%%%%: unterminated text
