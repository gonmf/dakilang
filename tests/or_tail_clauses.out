> actor("chaplin").
> athlete('ronaldo').
> athlete('messi').
> politician("ghandi").
> politician("arnold").
> politician(123).
> actor("arnold").
> athlete("arnold").
>
> famous(X) :- actor(X);               athlete(X);              politician(X).
>
> listing
 0: actor('chaplin').
 1: athlete('ronaldo').
 2: athlete('messi').
 3: politician('ghandi').
 4: politician('arnold').
 5: politician(123).
 6: actor('arnold').
 7: athlete('arnold').
 8: famous(X) :- actor(X).
 9: famous(X) :- athlete(X).
10: famous(X) :- politician(X).

>
> famous(X)?
X = 'chaplin'

X = 'ronaldo'

X = 'ghandi'

X = 'arnold'

X = 'messi'

X = 123

>
>
> other(X) :- abc(X, "gual").
> other(X) :- abc(X, "guali").
> other(X) :- abc(X, "gual").
>
> listing
 0: actor('chaplin').
 1: athlete('ronaldo').
 2: athlete('messi').
 3: politician('ghandi').
 4: politician('arnold').
 5: politician(123).
 6: actor('arnold').
 7: athlete('arnold').
 8: famous(X) :- actor(X).
 9: famous(X) :- athlete(X).
10: famous(X) :- politician(X).
11: other(X) :- abc(X, 'gual'). (2)
12: other(X) :- abc(X, 'guali').

>
> other(X) :- abc(X, "guali"); abc(X, "gual").
>
> ###########################
>
> parent('john', 'damian').
> parent('john', 'ellie').
> parent('sarah', 'john').
>
> grandparent(gp, gc) :- parent(gp, parent), parent(parent, gc).
>
> sibling(X, Y) :- grandparent(Z, X), grandparent(Z, Y).
>
> son(X, Y) :- parent(Y, X).
>
> related(person, other) :- son(parent, other); parent(parent, other); sibling(parent, other).
>
> related('sarah', anyone)?
anyone = 'damian'

anyone = 'john'

anyone = 'ellie'

anyone = 'sarah'

>
